### PR TITLE
[MIG] product_label_image: Migration to 15.0

### PR DESCRIPTION
**Pull request by nikunjantala**

### CODE REVIEW

**Review by ikapasi-initos — APPROVED** [review]
Code review and functional test.
LGTM

**Comment by jans23** [comment]
@nikunjantala Add the latest changes from 12.0.

**Comment by jans23** [comment]
Please create a PR to OCA.

**Comment by jans23** [comment]
any update?

**Comment by jans23** [comment]
Update this module with latest changes from 12.0 module version.

**Comment by nileshsheliya** [comment]
Module updated from latest changes from 12.0

File: product_label_image/reports/product_template_report.xml [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+                        <div class="text-center align-middle" style="height: 10rem;">
+                            <img t-att-src="'data:image/png;base64,%s' % to_text(product.image_512)"
+                                 style="height:109px;width:auto;"/>
+                        </div>
+                    </div>
+                </t>
+            </xpath>
+        </template>
+
+        <template id="report_simple_barcode_with_image" inherit_id="product.report_simple_label4x7">
```
Comment (alihashaam) [inline product_label_image/reports/product_template_report.xml]: duplicated template_id as line 4. Please use different id. OCA pylint would throw duplicate id error

File: product_label_image/reports/product_template_report.xml [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+                        <div class="text-center align-middle" style="height: 10rem;">
+                            <img t-att-src="'data:image/png;base64,%s' % to_text(product.image_512)"
+                                 style="height:109px;width:auto;"/>
+                        </div>
+                    </div>
+                </t>
+            </xpath>
+        </template>
+
+        <template id="report_simple_barcode_with_image" inherit_id="product.report_simple_label4x12">
```
Comment (alihashaam) [inline product_label_image/reports/product_template_report.xml]: duplicated template_id as line 4. Please use different id. OCA pylint would throw duplicate id error

File: product_label_image/reports/product_template_report.xml [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+                        <div class="text-center align-middle" style="height: 10rem;">
+                            <img t-att-src="'data:image/png;base64,%s' % to_text(product.image_512)"
+                                 style="height:63px;width:auto;"/>
+                        </div>
+                    </div>
+                </t>
+            </xpath>
+        </template>
+
+        <template id="report_simple_barcode_with_image" inherit_id="product.report_simple_label_dymo">
```
Comment (alihashaam) [inline product_label_image/reports/product_template_report.xml]: duplicated template_id as line 4. Please use different id. OCA pylint would throw duplicate id error

File: product_label_image/reports/product_template_report.xml [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+            <xpath expr="//div" position="after">
+                <t t-if="product.image_512">
+                    <div style="width: 97mm; border:1px solid #555555; height:37.1mm;" class="o_label_full">
+                        <div class="text-center align-middle" style="height: 10rem;">
+                            <img t-att-src="'data:image/png;base64,%s' % to_text(product.image_512)"
+                                 style="height:109px;width:auto;"/>
+                        </div>
+                    </div>
+                </t>
+            </xpath>
```
Comment (alihashaam) [inline product_label_image/reports/product_template_report.xml]: In the base module inside the template "report_simple_label2x7" and other inherited modules in this file, you can find the core functionality to display the Barcode. So, may be we can use that?

**Review by alihashaam — CHANGES_REQUESTED** [review]
@nikunjantala Please see the review

File: product_label_image/reports/product_template_report.xml (hunk 4 above)
Comment (nikunjantala) [reply product_label_image/reports/product_template_report.xml]: I have just updated code and added new option 4x6. So removed old template inheritance and created new fresh report for 4x6 size.

File: product_label_image/reports/product_template_report.xml [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+                  t-options="{'widget': 'barcode', 'symbology': 'auto', 'img_style': 'width:71mm;height:15mm;'}"
+                />
+                                <span class="text-center" t-out="barcode" />
+                            </t>
+                        </div>
+                    </div>
+                    <t t-if="product.image_512">
+                        <div
+              style="width:310px;height:210px; border:2px solid #555555;border-left:1px;"
+            >
```
Comment (manengel) [inline product_label_image/reports/product_template_report.xml]: To me it seems as if your code formatter does also make the code look very weird in cases there is no fixed technical ruleset regarding how it should look like.

File: product_label_image/reports/product_template_report.xml [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+                    <t t-set="barcode" t-value="barcode_and_qty[0]" />
+                    <t t-foreach="range(barcode_and_qty[1])" t-as="qty">
+                        <t t-call="product_label_image.report_simple_label_image_4x6" />
+                    </t>
+                </t>
+            </t>
+        </t>
+    </template>
+
+    <template id="producttemplatelabel_image_4x6">
```
Comment (manengel) [inline product_label_image/reports/product_template_report.xml:63]: Remaining duplicate.

**Review by manengel — CHANGES_REQUESTED** [review]
Still needs a bit of work.

File: product_label_image/reports/product_template_report.xml (hunk 6 above)
Comment (nileshsheliya) [reply product_label_image/reports/product_template_report.xml:63]: Duplicate ID removed already.

**Comment by alihashaam** [comment]
@nikunjantala what is the status here? is the PR ready to be reviewed? Because the last comment from Manuel says that it still needs some work and then there is no further update

File: product_label_image/reports/product_template_report.xml (hunk 5 above)
Comment (nileshsheliya) [reply product_label_image/reports/product_template_report.xml]: Its auto formatted based on pre-commit files given in the repo.

**Comment by alihashaam** [comment]
@nikunjantala 

Please let me know the status here as I would like to wrap up this PR

File: product_label_image/reports/product_template_report.xml (hunk 5 above)
Comment (nileshsheliya) [reply product_label_image/reports/product_template_report.xml]: @manengel Code auto formatted according to pre-commit files given in the repo. If we not follow it, PR failed. So we must need to follow it.

**Comment by nikunjantala** [comment]
@alihashaam As per Manuel code formatting is look very weird. But that formatting is done automatic based on pre-commit files and we must need to format it accordingly else PR will fail. So no need to make any changes in that. This PR is ready to review.

**Comment by alihashaam** [comment]
@jans23 

I have pushed two really small changes and now the review is done. This PR can be merged now

**Pull request merged**